parser-typechecker/src/Unison/Codebase.hs

```diff
@@ -197,14 +198,14 @@ addDefsToCodebase c uf = do
     goType f (ref, decl) = putTypeDeclaration c ref (f decl)
 
 getTypeOfConstructor ::
-  (Monad m, Ord v) => Codebase m v a -> Reference -> Int -> m (Maybe (Type v a))
-getTypeOfConstructor codebase (Reference.DerivedId r) cid = do
+  (Monad m, Ord v) => Codebase m v a -> ConstructorReference -> m (Maybe (Type v a))
+getTypeOfConstructor codebase (ConstructorReference (Reference.DerivedId r) cid) = do
```

Comment:
:man_shrugging: It's not super clear to me if we want that pattern synonym or not.

It also reminds me of the builtin suggestion from a week back:

```haskell
data Builtin a b
  = Builtin a 
  | UserDefined b
```

because what this code is really trying to do is switch on whether or not the thing's a builtin!

(...whoa, why do we have this if it error on builtins? we should just take a `ConstructorReferenceId` instead...)